### PR TITLE
iOS 11 bugs

### PR DESCRIPTION
**Pull request by nicorsm**
Visual & logic bugs on iOS 11, with support to iPhone X

### CODE REVIEW

**Comment by nicorsm** [comment]
- [X] Chat, header misalignment
- [X] Conversations list, iPhone X support
- [X] Profile view, iPhone X support
- [X] Chat, iPhone X support
- [X] Self profile view, iPhone X support
- [X] Registration/login workflow, fixes to logo & iPX
- [ ] Self profile view, settings button not visible

File: Wire-iOS/Sources/UserInterface/Conversation/ConversationRootViewController.swift [hunk 1]
```diff
@@ -61,19 +61,22 @@ import Cartography
...
-            
-            contentView.left == view.left
-            contentView.right == view.right
-            contentView.bottom == view.bottom
-            contentView.top == customNavBar.bottom
-            
-            conversationViewControllerView.edges == contentView.edges
-        }
+        
+            constrain(self.customNavBar, self.view, self.contentView, conversationViewController.view) { (customNavBar: LayoutProxy, view: LayoutProxy, contentView: LayoutProxy, conversationViewControllerView: LayoutProxy) -> () in
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/Conversation/ConversationRootViewController.swift]: unwanted indentation?

Comment (nicorsm) [reply Wire-iOS/Sources/UserInterface/Conversation/ConversationRootViewController.swift]: Probably 🙂

File: Wire-iOS/Sources/Helpers/UIScreen+SafeArea.swift [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+import UIKit
+
+extension UIScreen {
+    
+    static var safeArea: UIEdgeInsets {
+        if #available(iOS 11, *) {
+            if let window = UIApplication.shared.keyWindow {
+                let insets = window.safeAreaInsets
+                if insets.top > 0 {
```
Comment (mikeger) [inline Wire-iOS/Sources/Helpers/UIScreen+SafeArea.swift]: Not sure if the logic is clear enough here, I guess if there is a top inset this means that this is an iPhone X? I think the comment can be added.

**Comment by nicorsm** [comment]
The account view glitch was caused by the `edgesWithinMargin` constraint. Changing it to `edges` solved the problem.

**Comment by mikeger** [comment]
@nicorsm let's merge it for now and fix share extension in the separate PR?

**Comment by nicorsm** [comment]
@mikeger I was thinking the same!

**Pull request merged**